src/ast.hpp

```diff
@@ -228,37 +218,116 @@ namespace Sass {
   public:
     Vectorized(size_t s = 0) : hash_(0)
     { elements_.reserve(s); }
+    Vectorized(const std::vector<T> vec) : hash_(0)
```

Comment:
Since you're copying it here, there is no need for `const`, and you can move the vector in to avoid double copy:

```c++
Vectorized(std::vector<T> vec) : hash_(0), elements_(std::move(vec)) {}
```

Reply:
Is a move safe here? IMO we must make copies in this constructor?

Reply:
Yes, a move is perfectly safe here, because the `vec` vector is passed by value.

Reply:
Got it, made the prototype to accept a reference then doing the copy on the initializer.